app/src/main/java/com/example/toyproject/ui/myskin/MySkinFragment.kt

```diff
@@ -0,0 +1,137 @@
+package com.example.toyproject.ui.myskin
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.isVisible
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import com.example.toyproject.base.BaseFragment
+import com.example.toyproject.databinding.FragmentMySkinBinding
+import com.example.toyproject.util.Resource
+import com.example.toyproject.util.extensions.setAdapter
+import com.example.toyproject.viewmodel.MySkinViewModel
+import com.google.android.material.chip.Chip
+import com.irozon.alertview.AlertActionStyle
+import com.irozon.alertview.AlertStyle
+import com.irozon.alertview.AlertView
+import com.irozon.alertview.objects.AlertAction
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import javax.inject.Inject
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@AndroidEntryPoint
+class MySkinFragment : BaseFragment<FragmentMySkinBinding>(), MySkinAdapter.OnClickListener {
+
+    @Inject
+    lateinit var adapter: MySkinAdapter
+
+    private val viewModel: MySkinViewModel by viewModels()
+
+    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentMySkinBinding =
+        FragmentMySkinBinding::inflate
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setupUI()
+        setupItems()
+        setupDeleteItem()
+        setupSearch()
+        setUpChips()
+    }
+
+    private fun setupUI() {
+        setAdapter(binding.rvMySkin, adapter)
+        binding.searchView.setQuery("", false)
+    }
+
+    private fun setupDeleteItem() {
+        binding.imgDeleteAll.setOnClickListener {
+            val alert = AlertView("Delete Item All", "choice", AlertStyle.IOS)
+            alert.addAction(AlertAction("Delete All", AlertActionStyle.NEGATIVE) {
+                viewModel.deleteAllSkins()
+            })
+            alert.show(requireActivity() as AppCompatActivity)
+        }
+    }
+
+    private fun setupItems() {
+        viewModel.skinLiveData.observe(viewLifecycleOwner, Observer {
+            when (it) {
+                is Resource.Success -> {
+                    showProgress(false)
+                    showRecyclerView(true)
+                    adapter.submitList(it.data)
+                }
```

Comment:
```suggestion
        viewModel.skinLiveData.observe(viewLifecycleOwner, Observer { skinList ->
            when (skinList) {
                is Resource.Success -> {
                    showProgress(false)
                    showRecyclerView(true)
                    adapter.submitList(skinList.data)
                }
```
it 으로 받아와서 해당 코드들이 잘 읽히지 않는 것 같아요🥺. 명시적으로 선언해주는게 어떨까 하여 제안드려봅니다 : )